src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Charts/LineChart/LineChart.tsx

```diff
@@ -1,36 +1,95 @@
-import { useMemo } from 'react';
+import { useMemo, useRef, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { useFormContext } from 'react-hook-form';
-import {
-  Chart as ChartJS,
-  LinearScale,
-  CategoryScale,
-  PointElement,
-  LineElement,
-  Tooltip,
-  TimeScale,
-} from 'chart.js';
+import { Chart as ChartJS, Tooltip, TimeScale, Legend, ScriptableTooltipContext } from 'chart.js';
 import 'chartjs-adapter-date-fns';
 import ChartDataLabels from 'chartjs-plugin-datalabels';
 import { Line } from 'react-chartjs-2';
+import { ChartJSOrUndefined } from 'react-chartjs-2/dist/types';
 
+import { getOptionTextApi } from 'api';
+import { useAsync } from 'shared/hooks';
 import { FilterFormValues } from 'modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.types';
 import { useDatavizFilters } from 'modules/Dashboard/hooks';
 
+import { locales, TOOLTIP_OFFSET_LEFT, TOOLTIP_OFFSET_TOP } from '../Charts.const';
 import { getOptions, getData } from './LineChart.utils';
-import { CustomLegend, LineChartProps } from './LineChart.types';
-import { locales } from '../Charts.const';
+import { CustomLegend, DataPointRaw, LineChartProps, TooltipData } from './LineChart.types';
+import { ChartTooltip } from './ChartTooltip';
 
-ChartJS.register(LinearScale, CategoryScale, PointElement, LineElement, Tooltip, TimeScale);
+ChartJS.register(Tooltip, TimeScale, Legend);
 
 export const LineChart = ({ data, versions }: LineChartProps) => {
   const { i18n } = useTranslation('app');
+  const chartRef = useRef<ChartJSOrUndefined<'line', DataPointRaw[]> | null>(null);
 
+  const tooltipRef = useRef<HTMLDivElement | null>(null);
+  const isHovered = useRef(false);
+
+  const [tooltipData, setTooltipData] = useState<TooltipData[] | null>(null);
+
+  const { execute: getOptionText } = useAsync(getOptionTextApi);
   const { watch } = useFormContext<FilterFormValues>();
   const { minDate, maxDate, filteredVersions } = useDatavizFilters(watch, versions);
 
   const lang = i18n.language as keyof typeof locales;
 
+  const hideTooltip = () => {
+    const tooltipEl = tooltipRef.current;
+
+    if (!tooltipEl) return;
+
+    isHovered.current = false;
+    tooltipEl.style.display = 'none';
+  };
+
+  const tooltipHandler = async (context: ScriptableTooltipContext<'line'>) => {
+    const tooltipEl = tooltipRef.current;
+
+    if (!tooltipEl) return;
+
+    const { tooltip } = context;
+    const { dataPoints } = tooltip;
+
+    if (!tooltip.opacity && !isHovered.current) {
+      tooltipEl.style.display = 'none';
+
+      return;
+    }
+
+    const chart = chartRef.current;
+
+    if (chart && dataPoints.length) {
+      const tooltipDataPoints = await Promise.all(
+        dataPoints.map(async (dataPoint) => {
+          let optionText = (dataPoint.raw as DataPointRaw).optionText;
+          const linkPattern = new RegExp('^(http|https)://', 'i');
```

Comment:
maybe it's better to move it in `LineChart.const` as `new RegExp('^http(s?)://', 'i');`